src/libs/actions/Card.ts

```diff
@@ -0,0 +1,23 @@
+import * as API from '../API';
+import CONST from '../../CONST';
+import {TCardDetails} from '../../types/onyx/Card';
+import * as Localize from '../Localize';
+
+function revealVirtualCardDetails(cardID: string): Promise<TCardDetails> {
+    return new Promise((resolve, reject) => {
+        // eslint-disable-next-line rulesdir/no-api-side-effects-method
+        API.makeRequestWithSideEffects('RevealVirtualCardDetails', {cardID})
+            .then((response) => {
+                if (response.jsonCode !== CONST.JSON_CODE.SUCCESS) {
+                    reject(response.message || Localize.translateLocal('cardPage.cardDetailsLoadingFailure'));
```

Comment:
I think we should not return the `response.message` error. It could be anything. Same with the catch below. Just return the default translated error in either case since the user will end up seeing this.